### PR TITLE
Update to 1.21.3 for real, I think?

### PR DESCRIPTION
**Pull request by cputnam-a11y**
Updated to 1.21.3.
Added Recipes to datagen, because I didn't want to deal with it.
Fell in love with cicada.
Switched to official maven for cooldown coordinator rather than enjarai's mirror
Added and then removed ModItemTagGenerator.
Did not pull a stupid and get into a place where pr didn't work

### CODE REVIEW

File: src/main/resources/fabric.mod.json [hunk 1]
```diff
@@ -34,13 +31,13 @@
...
   ],
-  "accessWidener" : "omnihopper.accesswidener",
+  "accessWidener": "omnihopper.accesswidener",
   "depends": {
     "cooldown-coordinator": "^0.4.3",
     "cicada": ">=0.6.0 <1.0.0",
     "fabric": "*",
-    "fabricloader": ">=0.15",
-    "minecraft": ">=1.20.5 <=1.20.6",
+    "fabricloader": ">=0.16.9",
```
Comment (enjarai) [inline src/main/resources/fabric.mod.json:39]: Is there a reason to specifically demand a higher fabric loader version?

File: src/main/resources/assets/omnihopper/models/block/hopper_down.json
Comment (enjarai) [inline src/main/resources/assets/omnihopper/models/block/hopper_down.json:1]: Why reformat all the json files? Especially for models it seems kinda counterproductive.

File: src/main/resources/assets/omnihopper/lang/en_us.json
Comment (enjarai) [inline src/main/resources/assets/omnihopper/lang/en_us.json:1]: I keep empty lines in many of these files for a reason actually...

File: src/main/java/nl/enjarai/omnihopper/items/ModItems.java [hunk 2]
```diff
@@ -21,13 +22,15 @@ public class ModItems {
...
+    }
 
     private static BlockItem registerBlockItem(Block block) {
         Identifier id = Registries.BLOCK.getId(block);
-        var item = Registry.register(Registries.ITEM, id, new BlockItem(block, new Item.Settings()) {
+        var item = Registry.register(Registries.ITEM, id, new BlockItem(block, new Item.Settings().registryKey(keyOf(id)).useBlockPrefixedTranslationKey()) {
+
             @Override
-            public void appendTooltip(ItemStack stack, Item.TooltipContext context, List<Text> tooltip, TooltipType type) {
+            public void appendTooltip(ItemStack stack, TooltipContext context, List<Text> tooltip, net.minecraft.item.tooltip.TooltipType type) {
```
Comment (enjarai) [inline src/main/java/nl/enjarai/omnihopper/items/ModItems.java:33]: Why fully qualify the type?

File: src/main/java/nl/enjarai/omnihopper/blocks/OpenBoxBlock.java [hunk 3]
```diff
@@ -106,16 +109,27 @@ public boolean canFillWithFluid(@Nullable PlayerEntity player, BlockView world,
...
+//        return super.getStateForNeighborUpdate(state, direction, neighborState, world, pos, neighborPos);
+//    }
+
+
     @Override
-    public BlockState getStateForNeighborUpdate(BlockState state, Direction direction, BlockState neighborState, WorldAccess world, BlockPos pos, BlockPos neighborPos) {
+    protected BlockState getStateForNeighborUpdate(BlockState state, WorldView world, ScheduledTickView tickView, BlockPos pos, Direction direction, BlockPos neighborPos, BlockState neighborState, Random random) {
         if (state.get(WATERLOGGED)) {
-            world.scheduleFluidTick(pos, Fluids.WATER, Fluids.WATER.getTickRate(world));
+            System.out.println("is instance of ScheduledTickView" + (world instanceof ScheduledTickView));
```
Comment (enjarai) [inline src/main/java/nl/enjarai/omnihopper/blocks/OpenBoxBlock.java:124]: I don't think this is supposed to be here?

**Review by enjarai — COMMENTED** [review]
As a general question: why base this PR on the 1.20.5 version? A 1.21 version already existed, and this might cause conflicts when merging features down the line.

I've added a few other comments where I have specific questions.

File: src/main/resources/assets/omnihopper/lang/en_us.json
Comment (cputnam-a11y) [reply src/main/resources/assets/omnihopper/lang/en_us.json:1]: I didn't specifically remove the lines and reformat the model files manually. assuming this is a problem, I can just discard all changes to them, as I haven't made any functional changes to them.

File: src/main/java/nl/enjarai/omnihopper/blocks/OpenBoxBlock.java (hunk 3 above)
Comment (cputnam-a11y) [reply src/main/java/nl/enjarai/omnihopper/blocks/OpenBoxBlock.java:124]: Yeah. I'll fix that

File: src/main/java/nl/enjarai/omnihopper/items/ModItems.java (hunk 2 above)
Comment (cputnam-a11y) [reply src/main/java/nl/enjarai/omnihopper/items/ModItems.java:33]: Not manually once again, there were errors in the signature, so I commented it out and let idea Regen. Will fix.

File: src/main/resources/fabric.mod.json (hunk 1 above)
Comment (cputnam-a11y) [reply src/main/resources/fabric.mod.json:39]: Not specifically, I like to do it when updating mods, but I don't believe I've used any newer features

**Comment by cputnam-a11y** [comment]
> As a general question: why base this PR on the 1.20.5 version? A 1.21 version already existed, and this might cause conflicts when merging features down the line.

 Simply, the 1.21 version didn't show up in the branch switch menu on web? Will fix. Should be easier than the first time

**Pull request closed without merging**